codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectAggregator.java

```diff
@@ -335,48 +337,27 @@ public boolean release(int decrement) {
             super(request, content, trailingHeaders);
         }
 
-        /**
-         * Copy this object
-         *
-         * @param copyContent
-         * <ul>
-         * <li>{@code true} if this object's {@link #content()} should be used to copy.</li>
-         * <li>{@code false} if {@code newContent} should be used instead.</li>
-         * </ul>
-         * @param newContent
-         * <ul>
-         * <li>if {@code copyContent} is false then this will be used in the copy's content.</li>
-         * <li>if {@code null} then a default buffer of 0 size will be selected</li>
-         * </ul>
-         * @return A copy of this object
-         */
-        private FullHttpRequest copy(boolean copyContent, ByteBuf newContent) {
-            DefaultFullHttpRequest copy = new DefaultFullHttpRequest(
-                    protocolVersion(), method(), uri(),
-                    copyContent ? content().copy() :
-                            newContent == null ? Unpooled.buffer(0) : newContent);
-            copy.headers().set(headers());
-            copy.trailingHeaders().set(trailingHeaders());
-            return copy;
+        @Override
+        public FullHttpRequest copy() {
+            return replace(content().copy());
         }
 
         @Override
-        public FullHttpRequest copy(ByteBuf newContent) {
-            return copy(false, newContent);
+        public FullHttpRequest duplicate() {
+            return replace(content().duplicate());
         }
 
         @Override
-        public FullHttpRequest copy() {
-            return copy(true, null);
+        public FullHttpRequest retainedDuplicate() {
+            return replace(content().retainedDuplicate());
         }
 
         @Override
-        public FullHttpRequest duplicate() {
-            DefaultFullHttpRequest duplicate = new DefaultFullHttpRequest(
-                    getProtocolVersion(), getMethod(), getUri(), content().duplicate());
-            duplicate.headers().set(headers());
-            duplicate.trailingHeaders().set(trailingHeaders());
-            return duplicate;
+        public FullHttpRequest replace(ByteBuf content) {
+            DefaultFullHttpRequest dup = new DefaultFullHttpRequest(protocolVersion(), method(), uri(), content);
```

Comment:
not introduced by this PR but I just noticed that the new headers may end up validating headers when the current headers do not (and visa versa). Should I open a new issue for this?